lib/rules/match-component-file-name.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * @fileoverview Require component name property to match its file name
+ * @author Rodrigo Pedra Brum <rodrigo.pedra@gmail.com>
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+const path = require('path')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'require component name property to match its file name',
+      category: undefined,
+      url: 'https://github.com/vuejs/eslint-plugin-vue/blob/v5.0.0-beta.4/docs/rules/match-component-file-name.md'
+    },
+    fixable: null,
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          extensions: {
+            type: 'array',
+            items: {
+              type: 'string'
+            },
+            uniqueItems: true,
+            additionalItems: false
+          }
+        },
+        additionalProperties: false
+      }
+    ]
+  },
+
+  create (context) {
+    const options = context.options[0]
+    const extensionsArray = options && options.extensions
+    const allowedExtensions = Array.isArray(extensionsArray) ? extensionsArray : ['jsx']
+    const [, filename, extension] = /^(.+?)\.(.+)$/g.exec(path.basename(context.getFilename()))
```

Comment:
Looks like crashing when the filename doesn't have that file extension.

(we can use `stdin` to lint (E.g., `curl a_url | eslint --stdin`), in such a case, the filename is `"<INPUT>"`).

How about?:

```suggestion
    const extension = path.extname(context.getFilename())
    const filename = path.basename(context.getFilename(), extension)
```

Reply:
The `extname` method keeps the leading dot in the extension. Should I trim the dot before comparing, or do you think it is a better idea to ask the users to specify the extension with a dot in the options?

Reply:
Good question. I don't have a strong opinion here.

For example, an ESLint CLI option need the beginning dot: `eslint src --ext .js,.vue`.
But I know another case that doesn't need the beginning dot.

I'm OK with either way.

Reply:
I kept as it was, without the leading dot. As it only compares once per file I don't think it is a performance hit.